ipv8/test/REST/rtest/peer_interaction_test_env.py

```diff
@@ -0,0 +1,485 @@
+        {
+            'port': the master peer's port. Defaults to 8086
+            '8086': the master peer's path to the working directory. Defaults to 'test_env'
+            'interface': IP or alias of the peer. Defaults to '127.0.0.1'
+            'configuration': IPv8 configuration object. Defaults to None
+            'get_style_requests': GET style request generator. Defaults to None
+            'post_style_requests': POST style request generator. Defaults to None
+        }
+        """
+
+        port = kwargs.get('port', 8086)
+        path = kwargs.get('path', 'test_env')
+        interface = kwargs.get('interface', '127.0.0.1')
+        configuration = kwargs.get('configuration', None)
+        get_style_requests = kwargs.get('get_style_requests', None)
+        post_style_requests = kwargs.get('post_style_requests', None)
+
+        # Create a so called master (well-known) peer, which should be the peer to which the requests are directed
+        self._master_peer = TestPeer(path, port, interface, configuration)
+
+        # Check to see if the user has provided request generators
+        if get_style_requests:
+            assert isinstance(get_style_requests, GetStyleRequests), "The get_style_requests parameter must be a " \
+                                                                     "subclass of GetStyleRequests"
+            self._get_style_requests = get_style_requests
+        else:
+            # If no get style request provided, default to the HTTP implementation
+            self._get_style_requests = HTTPGetRequester()
+
+        if post_style_requests:
+            assert isinstance(post_style_requests, PostStyleRequests), "The post_style_requests parameter must be a " \
+                                                                       "subclass of PostStyleRequests"
+            self._post_style_requests = post_style_requests
+        else:
+            # If no post style request provided, default to the HTTP implementation
+            self._post_style_requests = HTTPPostRequester()
+
+    def setUp(self):
+        # Call the method which sets up the environment
+        self.initialize()
+
+    def tearDown(self):
+        # Call super method
+        super(SingleServerSetup, self).tearDown()
+
+        # Stop the master peer
+        self._master_peer.stop()
+
+
+class RequestTest(SingleServerSetup):
+
+    @inlineCallbacks
+    def wait_for_peers(self, dict_param):
+        """
+        Wait until this peer receives a non-empty list of fellow peers in the network
+
+        :param dict_param: the required parameters by the GET request generator for the peers request type
+        :return: a list of currently known peers in the network
+        """
+
+        peer_list = yield self._get_style_requests.make_peers(dict_param)
+
+        # Keep iterating until peer_list is non-empty
+        while peer_list == "[]":
+            # Wait for 4 seconds before trying again
+            time.sleep(4)
```

Comment:
This does not do what you think it does while running on the Twisted thread. By sleeping you actually block the entirety of Python and almost nothing else will happen. Use `yield deferLater(reactor, time, lambda: None)` instead.